resources/lib/ui/control.py

```diff
@@ -104,3 +120,37 @@ def draw_items(video_data):
             xbmc_add_player_item(vid['name'], vid['url'], vid['image'])
     xbmcplugin.endOfDirectory(HANDLE, succeeded=True, updateListing=False, cacheToDisc=True)
     return True
+
+
+def log(msg, level="INFO"):
+    log_message = '{0}: {1}'.format(ADDON_NAME, msg)
+    if level == "DEBUG":
+        xbmc.log(msg=log_message, level=xbmc.LOGDEBUG)
+    if level == "INFO":
+        xbmc.log(msg=log_message, level=xbmc.LOGINFO)
+    if level == "NOTICE":
+        xbmc.log(msg=log_message, level=xbmc.LOGNOTICE)
+    if level == "WARNING":
+        xbmc.log(msg=log_message, level=xbmc.LOGWARNING)
+    if level == "ERROR":
+        xbmc.log(msg=log_message, level=xbmc.LOGERROR)
+    if level == "SEVERE":
+        xbmc.log(msg=log_message, level=xbmc.LOGSEVERE)
+    if level == "FATAL":
+        xbmc.log(msg=log_message, level=xbmc.LOGFATAL)
+
+
+def aria2_download(url, fname, path):
+    aria_url = "http://{}:{}/rpc".format(getSetting('ipaddress'), getSetting('port'))
```

Comment:
to solve the playSource() thingy you want to add here:
linkInfo = _prefetch_play_link(url)

then use linkInfo['url']